mautic/mautic
# Notify user that an email is published/not published in a campaign

**Please be sure you are submitting this against the _staging_ branch.**

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | N
| New feature? | Y
| Automated tests included? | Y
| Related user documentation PR URL | 
| Related developer documentation PR URL | 
| Issues addressed (#s or URLs) | 
| BC breaks? | N
| Deprecations? | Y

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:
Is good to be notified when working with campaign which has unpublished e-mails.

[//]: # ( As applicable: )
#### Steps to reproduce the bug:
1. Publish unpublished campaign with unpublished email. No notification is shown.

#### Steps to test this PR:
1. Publish unpublished campaign with unpublished email.
2. Edit unpublished campaign with unpublished email. (Notifications on frontend does not support multiple notifications, so only one is shown - without "successfully saved")

#### List deprecations along with the new alternative:
1.  app/bundles/CoreBundle/Controller/CommonController.php::addNotification() - only proxy for now
2. app/bundles/CoreBundle/Controller/CommonController.php::addFlash() - only proxy for now
3. app/bundles/CoreBundle/Controller/CommonController.php::addBrowserNotification() - unused

#### List backwards compatibility breaks:
None.
